lib/google_scraping_web/controllers/api/v1/upload_keyword_controller.ex

```diff
@@ -0,0 +1,76 @@
+defmodule GoogleScrapingWeb.Api.V1.UploadKeywordController do
+  use GoogleScrapingWeb, :controller
+
+  alias GoogleScraping.Dashboard.Keywords
+  alias GoogleScraping.Dashboard.Schemas.{Keyword, KeywordCSVFile}
+  alias GoogleScrapingWeb.Controllers.ErrorHandler
+
+  def create(conn, %{"keyword_csv_file" => params}) do
+    changeset = %{
+      KeywordCSVFile.create_changeset(%KeywordCSVFile{}, %{file: params})
+      | action: :validate
+    }
+
+    with %Ecto.Changeset{valid?: true, changes: %{file: file}} <- changeset,
+         {:ok, keyword_list} <- KeywordCSVFile.parse(file.path) do
+      Keywords.create_keyword_list(keyword_list, conn.assigns.current_user)
+
+      conn
+      |> put_status(:created)
+      |> render("show.json", %{
+        data: %{
+          id: :os.system_time(:millisecond),
+          message: "Keywords were uploaded!"
+        }
+      })
+    else
+      %Ecto.Changeset{valid?: false} ->
+        changeset_errors = ErrorHandler.build_changeset_error_message(changeset)
+
+        conn
+        |> put_status(:bad_request)
+        |> ErrorHandler.render_error_json(:bad_request, changeset_errors)
+
+      {:error, reason} ->
+        process_validation_error(conn, reason)
+    end
+  end
+
+  def create(conn, _params) do
+    ErrorHandler.render_error_json(
+      conn,
+      :unprocessable_entity,
+      "Invalid input attributes. Add `keyword_csv_file` to request body"
+    )
+  end
+
+  defp process_validation_error(conn, reason) do
```

Comment:
It it possible to move this function in a separate module? as this one is quite big and makes the controller fat. may be in `ErrorHandler` or other separate module?

Reply:
I'm not sure that it is worth moving full error handling to a separate module. It is directly related to the logic of the controller - validation of user input.

I refactored this method a bit. Now it looks better: https://github.com/Nihisil/nimble-certification-elixir/pull/55/commits/0ec08506524cbb09538bb112231e1319fffb292f